### PR TITLE
fix: enforce Postgres unique index ownership

### LINKED ISSUE #95: [P0] Postgres adapter does not enforce unique indexes despite claiming atomic support
The Postgres adapter declares `capabilities.uniqueConstraints = "atomic"`, so the store will trust the adapter to reject duplicate values for `unique: true` indexes. The adapter does not appear to store or enforce unique index ownership anywhere, so duplicate values can be written without an `EngineUniqueConstraintError`.

Evidence:
- `src/engines/postgres.ts:164` declares atomic unique support.
- The adapter schema only creates document/index/migration tables; there is no unique-index ownership table.
- The write-path methods (`create`, `put`, `update`, `batchSet`) do not accept or process the `uniqueIndexes` payload.
- The file has no `EngineUniqueConstraintError` branch and no unique probe path.

Impact:
- Applications can assume uniqueness is protected when it is not.
- Duplicate values will surface later as data-integrity bugs rather than immediate write failures.

Suggested fix:
- Add adapter-level unique-index storage and transactional enforcement for Postgres writes, or downgrade the capability flag until that exists.
- Add integration tests that prove duplicate create/update/batchSet attempts fail for Postgres-backed models with unique indexes.

### PR DESCRIPTION
**Pull request by samlaycock**
## Summary
- add a dedicated Postgres unique-index ownership table and provision it during schema bootstrap
- enforce unique index ownership across `create`, `put`, `update`, `batchSet`, and `batchSetWithResult`, translating duplicate writes into `EngineUniqueConstraintError`
- enable the shared Postgres unique-constraint conformance coverage and cover custom naming for the new internal table

## Testing
- `bun run fmt`
- `bun run lint:fix`
- `bun run lint` *(passes with pre-existing warnings in `tests/integration/conformance-suite.ts` and `tests/unit/sqlite-engine.test.ts`)*
- `bun run test`
- `bun run typecheck`
- `bun run services:up:postgres` *(blocked in this environment: Docker daemon unavailable at `/Users/samuellaycock/.orbstack/run/docker.sock`)*

Closes #95.

### CODE REVIEW

**Comment by samlaycock** [comment]
@greptile Take another look.

File: src/engines/postgres.ts [hunk 1]
```diff
@@ -1012,7 +1420,7 @@ async function applyBatchSetChunk(
 async function ensureSchema(client: PostgresQueryableLike, refs: TableRefs): Promise<void> {
   const session = await acquireSession(client);
   const lockKey = toAdvisoryLockKey(
-    `${refs.schema}|${refs.documentsTable}|${refs.indexesTable}|${refs.migrationMetadataTable}|${refs.migrationLocksTable}|${refs.migrationCheckpointsTable}`,
+    `${refs.schema}|${refs.documentsTable}|${refs.indexesTable}|${refs.uniqueIndexesTable}|${refs.migrationMetadataTable}|${refs.migrationLocksTable}|${refs.migrationCheckpointsTable}`,
   );
```
[the remark this answers is not in the document]
Comment (samlaycock) [reply src/engines/postgres.ts]: Addressed in e0b0c9c. ensureSchema() now uses the pre-existing stable advisory-lock key again, with an explicit comment documenting why the lock domain must stay version-stable across rolling upgrades. That keeps mixed-version instances coordinating DDL while preserving the new unique index table bootstrap.
